File: tests/helpers/recipe-loader.js
const fs = require("node:fs");
const path = require("node:path");
const vm = require("node:vm");

// recipes.js を実行し、登録関数に渡される職人別レシピを取得します。
function loadFallbackRecipes(craftId) {
  const context = {
    recipes: null,
  };
  context.registerDQ10CraftRecipes = (registeredCraftId, recipes) => {
    if (registeredCraftId === craftId) {
      context.recipes = recipes;
    }
  };

  const recipesPath = path.join("app", "crafts", craftId, "recipes.js");
  vm.createContext(context);
  vm.runInContext(fs.readFileSync(recipesPath, "utf8"), context, { filename: recipesPath });

  if (!Array.isArray(context.recipes)) {
    throw new Error(`${craftId} のフォールバックレシピを読み込めません`);
  }

  return context.recipes;
}

module.exports = { loadFallbackRecipes };

File: tests/helpers/recipe_loader.py
"""コミット対象のフォールバックレシピを読み込みます。"""

import json
import re
from pathlib import Path


APP_CRAFTS_DIR = Path(__file__).resolve().parents[2] / "app" / "crafts"


def load_fallback_recipes(craft_id: str) -> list[dict]:
	"""recipes.js の登録呼び出しから職人別のレシピ配列を取り出します。"""
	contents = (APP_CRAFTS_DIR / craft_id / "recipes.js").read_text(encoding="utf-8")
	match = re.fullmatch(
		rf'\s*registerDQ10CraftRecipes\(\s*"{re.escape(craft_id)}"\s*,\s*(\[.*\])\s*\);\s*',
		contents,
		re.DOTALL,
	)
	if match is None:
		raise ValueError(f"{craft_id} のフォールバックレシピ形式が不正です")
	return json.loads(match.group(1))
